### ISSUE TITLE
OTA update over 0.5.1-dev not working on any device

### ISSUE THREAD

**Issue by nospam2k** [stelgenhof/AiLight#65]
Not sure if it's just me, but I have 8 light bulbs with 0.5.1-dev that work fine but I can't upgrade to .6 no matter what I do. I've even tried making an OTA stub but the upload fails (as does .6) with various error codes. Sometimes 512, 1024, or longer various strings of numbers. I was able to flash over one bulb using the pins but it's a real hassle as my jig is very finicky and takes a few tries to get it work.

**Comment by stelgenhof** [comment]
Could you maybe share a log file that shows some errors or other information?

Thanks! Sacha

**Comment by nospam2k** [comment]
I gave up and reflashed all of them to .6. I'll leave you to close this if you want.

**Comment by stelgenhof** [comment]
I would be still interested in the log file. OTA can be finicky so am keen to investigate this.

**Comment by nospam2k** [comment]
I'm sorry I don't have any log files. I was going to try and setup a test using a 4meg nodemcu. I did get an error on the OTA but it still uploaded and doing 2nd OTA was successful. I think being 4meg helped so maybe a 1meg chip test may show the problem. What ends up happening on the normal wifi light is the OTA errors and the original firmware is still there but the settings are corrupted. MQTT commands seem to still work. Settings cannot be reset or changed from the web interface. I have attached the firmware bin if you want to test it. It should be easily reproduced. I doesn't seem that any firmware can be OTA'd to the this firmware.

[AiLight.bin.zip](https://github.com/stelgenhof/AiLight/files/3962121/AiLight.bin.zip)

**Comment by stelgenhof** [comment]
Understood. Thank you for the file anyway. Not sure when, but will do some testing in the upcoming weeks :)

**Comment by stelgenhof** [comment]
@nospam2k Just a small note: Have you tried to do an OTA with a different code/script (a simple blink for example). For OTA to work, the free space needs to equal or more than the current firmware/script. In case it is larger, OTA simply will not work. It may be possible the AiLight firmware is too large :(

**Comment by nospam2k** [comment]
Yes. In my OP I said I tried making an "OTA stub".

**Comment by stelgenhof** [comment]
@nospam2k Ah yes... you did :)

**Comment by stelgenhof** [comment]
@nospam2k Did some tests with the binary you provided as well as my own code (0.5.1, 0.6, 0.6.1-dev) and was able to do an OTA upload without any issue.

![ota_ailight](https://user-images.githubusercontent.com/529544/71432189-00d85800-271b-11ea-8105-6bc18d2c8ae1.png)

Tested it on a lightbulb as well as on a ESP8266 board. As for the lightbulb, this had 0.5.1-dev installed and I uploaded the latest 0.6.1-dev code (like in your situation).

So not sure why yours fail to upload.
